### PR TITLE
[Feature] Added slotting generator.

### LINKED ISSUE #131: [Feature Request] Slotting website JSON generator
### **Is your feature request related to a problem? Please describe.**
I've got a proof-of-concept framework component that generates website json for a single squad, but it needs some stuff to be rigged up - it needs to run in multiplayer and you need to edit `description.ext` to set `disabledAI = false`.

### **Describe the solution you'd like**
The above to work without the specific rigging required, or if impossible, at least easy to do.
The above to generate the entire JSON file at once, so you don't need to go chasing squad members to extract the json from their bone marrow.

### **Why is this this best way to solve the problem?**
imagine writing a slotting json by hand very cringe

### **Additional context**
🤓

### PR DESCRIPTION
**Pull request by Bubbus**
Resolves #131 

### Pull Request Description
**When merged this pull request will:**
- Add a generator for slotting website JSON into the framework.
- Add a ZEN module to run the slotting generator and a configuration file to turn it on and off.

### Release Notes
Added a way to automatically generate slotting code for the mission slotting website!  To learn more, see the new `slottingGenerator.hpp` file in the `configuration` folder.

## IMPORTANT

- [x] Testing has been completed as neccessary, depending on the nature & impact of the changes.
- [x] The Release Notes section below must be filled out appropriately to explain the changes made in this PR. This section will be used in Framework Changelogs.
- [ ] If the contribution affects [the wiki](https://github.com/CombinedArmsGaming/CAFE3/wiki), please include your changes in this pull request so the wiki is consistently updated.
- [x] [Contribution Guidelines](https://github.com/CombinedArmsGaming/CAFE3/blob/release/contributing.md) are read, understood and applied.
- [x] Title of this PR uses our standard template `[Descriptor] - Add|Fix|Improve|Change|Make|Remove {changes}`.

### CODE REVIEW

File: components/slottingGenerator/fn_discoverAllUnitDLCs.sqf
Comment (Cre8or) [inline components/slottingGenerator/fn_discoverAllUnitDLCs.sqf:1]: Is this function intentionally blank?

File: components/slottingGenerator/fn_composeJSONUnknownFragments.sqf [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+	case "ARRAY": 
+	{
+		if (count _structure <= 0) exitWith {["[]"]};
+		private _sampleContent = _structure#0;
+
+		switch (typeName _sampleContent) do 
+		{
+			case "ARRAY": 
+			{
+				if ((count _sampleContent == 2) and ((typeName (_sampleContent#0)) == "STRING")) exitWith
```
Comment (Cre8or) [inline components/slottingGenerator/fn_composeJSONUnknownFragments.sqf]: Use `isEqualType` over manual type comparisons: 

```suggestion
				if ((count _sampleContent == 2) and {(_sampleContent#0) isEqualType ""}) exitWith 
```

Also note the use of `{}` for [short-circuit evaluation](https://community.bistudio.com/wiki/a_and_b), to ensure that we are always dereferencing an array index that is valid.

**Review by Cre8or — CHANGES_REQUESTED** [review]
Overall a very useful QoL feature; it took me some time to figure it out (as I've never dealt with slotting data before) but it seems to work as expected. Only noteworthy issue I found is the DLC detection stub - not sure whether that's intentional, an oversight or still a WIP.

Also, I'm not sure if this has been discussed, but what are your thoughts on modifying the `mission.sqm` to support the slotting generator module out-of-the-box? As it stands, users still have to:
- mark the overall commander (using `this setVariable ["f_leaderGroup", true]`), and
- link all of their groups together (using the "Sync to" feature)

for this PR to work. We can make their life easier by doing all of this ahead of time. This will gain even more importance once we add the other side presets back in (OPFOR/INDFOR slots). Maybe something to add to #211?

File: components/slottingGenerator/fn_discoverAllUnitDLCs.sqf
Comment (Bubbus) [reply components/slottingGenerator/fn_discoverAllUnitDLCs.sqf:1]: Yep, it's possible with some of the funny new funcs and config lookups, just never got around to it.  I'll throw this into a new ticket cause it's useful to have.

**Comment by Bubbus** [comment]
> Also, I'm not sure if this has been discussed, but what are your thoughts on modifying the `mission.sqm` to support the slotting generator module out-of-the-box?

Yep, that's what #211 is talking about.  When that was written, both PRs were just kinda vibing next to eachother so I couldn't do anything with the sqm cause sqm files do not play well with merge.

File: components/slottingGenerator/fn_discoverAllUnitDLCs.sqf
Comment (Bubbus) [reply components/slottingGenerator/fn_discoverAllUnitDLCs.sqf:1]: https://github.com/CombinedArmsGaming/CAFE3/issues/221

**Review by Cre8or — APPROVED** [review]
Works as expected. The DLC detection can be added later, I'm sure we can live without it for now.

:ship:

**Pull request merged**